<h1>Interface NamingEnumeration&lt;T&gt;</h1>     <dl> <dt>All Superinterfaces:</dt> <dd>
<a href="../../java/util/enumeration">Enumeration</a>&lt;T&gt;</dd> </dl>   <pre data-language="java">public interface NamingEnumeration&lt;T&gt;
extends Enumeration&lt;T&gt;</pre> <div>
<p>This interface is for enumerating lists returned by methods in the javax.naming and javax.naming.directory packages. It extends Enumeration to allow as exceptions to be thrown during the enumeration. </p>
<p> When a method such as list(), listBindings(), or search() returns a NamingEnumeration, any exceptions encountered are reserved until all results have been returned. At the end of the enumeration, the exception is thrown (by hasMore()); </p>
<p> For example, if the list() is returning only a partial answer, the corresponding exception would be PartialResultException. list() would first return a NamingEnumeration. When the last of the results has been returned by the NamingEnumeration's next(), invoking hasMore() would result in PartialResultException being thrown. </p>
<p> In another example, if a search() method was invoked with a specified size limit of 'n'. If the answer consists of more than 'n' results, search() would first return a NamingEnumeration. When the n'th result has been returned by invoking next() on the NamingEnumeration, a SizeLimitExceedException would then thrown when hasMore() is invoked. </p>
<p> Note that if the program uses hasMoreElements() and nextElement() instead to iterate through the NamingEnumeration, because these methods cannot throw exceptions, no exception will be thrown. Instead, in the previous example, after the n'th result has been returned by nextElement(), invoking hasMoreElements() would return false. </p>
<p> Note also that NoSuchElementException is thrown if the program invokes next() or nextElement() when there are no elements left in the enumeration. The program can always avoid this exception by using hasMore() and hasMoreElements() to check whether the end of the enumeration has been reached. </p>
<p> If an exception is thrown during an enumeration, the enumeration becomes invalid. Subsequent invocation of any method on that enumeration will yield undefined results.</p>
</div> <dl> <dt>Since:</dt> <dd>1.3</dd> <dt>See Also:</dt> <dd>
<a href="context#list-javax.naming.Name-"><code>Context.list(javax.naming.Name)</code></a>, <a href="context#listBindings-javax.naming.Name-"><code>Context.listBindings(javax.naming.Name)</code></a>, <a href="directory/dircontext#search-javax.naming.Name-javax.naming.directory.Attributes-java.lang.String:A-"><code>DirContext.search(javax.naming.Name, javax.naming.directory.Attributes, java.lang.String[])</code></a>, <a href="directory/attributes#getAll--"><code>Attributes.getAll()</code></a>, <a href="directory/attributes#getIDs--"><code>Attributes.getIDs()</code></a>, <a href="directory/attribute#getAll--"><code>Attribute.getAll()</code></a>
</dd> </dl>       <h2 id="method.detail">Methods</h2>   <h3 id="next--">next</h3> <pre data-language="java">T next()
throws NamingException</pre> <div>
<p>Retrieves the next element in the enumeration. This method allows naming exceptions encountered while retrieving the next element to be caught and handled by the application. </p>
<p> Note that <code>next()</code> can also throw the runtime exception NoSuchElementException to indicate that the caller is attempting to enumerate beyond the end of the enumeration. This is different from a NamingException, which indicates that there was a problem in obtaining the next element, for example, due to a referral or server unavailability, etc.</p>
</div> <dl> <dt>Returns:</dt> <dd>The possibly null element in the enumeration. null is only valid for enumerations that can return null (e.g. Attribute.getAll() returns an enumeration of attribute values, and an attribute value can be null).</dd> <dt>Throws:</dt> <dd>
<code><a href="namingexception">NamingException</a></code> - If a naming exception is encountered while attempting to retrieve the next element. See NamingException and its subclasses for the possible naming exceptions.</dd> <dd>
<code><a href="../../java/util/nosuchelementexception">NoSuchElementException</a></code> - If attempting to get the next element when none is available.</dd> <dt>See Also:</dt> <dd><a href="../../java/util/enumeration#nextElement--"><code>Enumeration.nextElement()</code></a></dd> </dl>    <h3 id="hasMore--">hasMore</h3> <pre data-language="java">boolean hasMore()
         throws NamingException</pre> <p>Determines whether there are any more elements in the enumeration. This method allows naming exceptions encountered while determining whether there are more elements to be caught and handled by the application.</p> <dl> <dt>Returns:</dt> <dd>true if there is more in the enumeration ; false otherwise.</dd> <dt>Throws:</dt> <dd>
<code><a href="namingexception">NamingException</a></code> - If a naming exception is encountered while attempting to determine whether there is another element in the enumeration. See NamingException and its subclasses for the possible naming exceptions.</dd> <dt>See Also:</dt> <dd><a href="../../java/util/enumeration#hasMoreElements--"><code>Enumeration.hasMoreElements()</code></a></dd> </dl>    <h3 id="close--">close</h3> <pre data-language="java">void close()
    throws NamingException</pre> <div>
<p>Closes this enumeration. After this method has been invoked on this enumeration, the enumeration becomes invalid and subsequent invocation of any of its methods will yield undefined results. This method is intended for aborting an enumeration to free up resources. If an enumeration proceeds to the end--that is, until <code>hasMoreElements()</code> or <code>hasMore()</code> returns <code>false</code>-- resources will be freed up automatically and there is no need to explicitly call <code>close()</code>. </p>
<p> This method indicates to the service provider that it is free to release resources associated with the enumeration, and can notify servers to cancel any outstanding requests. The <code>close()</code> method is a hint to implementations for managing their resources. Implementations are encouraged to use appropriate algorithms to manage their resources when client omits the <code>close()</code> calls.</p>
</div> <dl> <dt>Throws:</dt> <dd>
<code><a href="namingexception">NamingException</a></code> - If a naming exception is encountered while closing the enumeration.</dd> <dt>Since:</dt> <dd>1.3</dd> </dl>     <div class="_attribution">
  <p class="_attribution-p">
    &copy; 1993&ndash;2017, Oracle and/or its affiliates. All rights reserved.<br>Documentation extracted from Debian's OpenJDK Development Kit package.<br>Licensed under the GNU General Public License, version 2, with the Classpath Exception.<br>Various third party code in OpenJDK is licensed under different licenses (see Debian package).<br>Java and OpenJDK are trademarks or registered trademarks of Oracle and/or its affiliates.<br>
    
  </p>
</div>
